fem/hyperbolic.hpp

```diff
@@ -222,28 +361,55 @@ class HyperbolicFormIntegrator : public NonlinearFormIntegrator
                               const Vector &elfun, Vector &elvect) override;
 
    /**
-    * @brief implement <-hat(F)(u,x) n, [[v]]> with abstract hat(F) computed by
-    * ComputeFluxDotN and numerical flux object
+    * @brief Implements (J(u), ∇v) with abstract J computed by
+    * FluxFunction::ComputeFluxJacobian()
+    *
+    * @param[in] el local finite element
+    * @param[in] Tr element transformation
+    * @param[in] elfun local coefficient of basis
+    * @param[out] grad evaluated Jacobian
+    */
+   void AssembleElementGrad(const FiniteElement &el,
+                            ElementTransformation &Tr,
+                            const Vector &elfun, DenseMatrix &grad) override;
+
+   /**
+    * @brief Implements <-F̂(u,x) n, [v]> with abstract F̂ computed by
+    * RiemannSolver::Eval() of the numerical flux object
     *
     * @param[in] el1 finite element of the first element
     * @param[in] el2 finite element of the second element
     * @param[in] Tr face element transformations
     * @param[in] elfun local coefficient of basis from both elements
-    * @param[out] elvect evaluated dual vector <-hat(F)(u,x) n, [[v]]>
+    * @param[out] elvect evaluated dual vector <-F̂(u,x) n, [v]>
     */
    void AssembleFaceVector(const FiniteElement &el1,
                            const FiniteElement &el2,
                            FaceElementTransformations &Tr,
                            const Vector &elfun, Vector &elvect) override;
 
+   /**
+    * @brief Implements <-Ĵ(u,x) n, [v]> with abstract Ĵ computed by
+    * RiemannSolver::Grad() of the numerical flux object
+    *
+    * @param[in] el1 finite element of the first element
+    * @param[in] el2 finite element of the second element
+    * @param[in] Tr face element transformations
+    * @param[in] elfun local coefficient of basis from both elements
+    * @param[out] elmat evaluated Jacobian matrix <-Ĵ(u,x) n, [v]>
+    */
+   void AssembleFaceGrad(const FiniteElement &el1,
+                         const FiniteElement &el2,
+                         FaceElementTransformations &Tr,
+                         const Vector &elfun, DenseMatrix &elmat) override;
 };
 
 
 /**
  * @brief Rusanov flux, also known as local Lax-Friedrichs,
  *    F̂ n = ½(F(u⁺,x)n + F(u⁻,x)n) - ½λ(u⁺ - u⁻)
  * where λ is the maximum characteristic velocity
- *
+ * @note The implementation assumes monotonous F(u,x) in u
```

Comment:
This statement doesn't generalize to systems... perhaps it can be moved to the particular part of the implementation that assumes monotonicity with scalar equations?

Reply:
Sounds like what you want to say is that the flux is "convex" instead of "monotonous"?  Technically, the proper way to document this would be to just update the documentation to say that lambda is `max(|F(u-,x)n|, |F(u+,x)n|)` and not that it's the maximum characteristic velocity... because those two are not generally equivalent.  This would avoid trying to determine under what conditions `max(|F(u-,x)n|, |F(u+,x)n|)` is the maximum characteristic speed.

Reply:
Ok, I put the exact definition to the brief description and explanation to the note. However, it is important to explain to the user under what assumptions it is Rusanov flux actually and when it is provably stable, because you do not want to deduce that by yourself from the implementation (or its description). So it is Rusanov flux simplified under the assumption of a monotonous flux (as the endpoints are arbitrary, only a monotonous function is convex in all cases).

Reply:
Burgers flux $f(u) = \frac{1}{2} u^2$ is not monotonic.  As I suggested in another comment, perhaps "convex" is the appropriate property?

Reply:
I took another look through the implementation and the statement

>  The reason is that lambda is taken as max(|F(u-,x)n|, |F(u+,x)n|)

is not correctly capturing how lambda is currently implemented.  Given this thread is on an outdated version of the documentation, I'll elaborated on the updated version.